NGitLab.Mock/Clients/GroupClient.cs

```diff
@@ -155,28 +122,66 @@ public IEnumerable<Models.Group> Get(GroupQuery query)
         }
     }
 
-    public GitLabCollectionResponse<Models.Group> GetAsync(GroupQuery query)
+    public GitLabCollectionResponse<Models.Group> GetAsync(GroupQuery query) => GitLabCollectionResponse.Create(Get(query));
+
+    public async Task<(IReadOnlyCollection<Models.Group> Page, int? Total)> PageAsync(PageQuery<GroupQuery> query, CancellationToken cancellationToken = default)
     {
-        return GitLabCollectionResponse.Create(Get(query));
+        await Task.Yield();
+        var pageNum = Math.Max(1, query?.Page ?? PageQuery.FirstPage);
+        var perPage = query?.PerPage ?? PageQuery.DefaultPerPage;
+        if (perPage < PageQuery.MinPerPage)
+        {
+            // Max isn't enforced the same way
+            throw new GitLabBadRequestException("per_page has a value not allowed");
```

Comment:
```suggestion
            throw new GitLabBadRequestException($"per_page value ({perPage}) is invalid: cannot be lower than MinPerPage ({PageQuery.MinPerPage})");
```

To give more context.

Reply:
@Toa741 I don't have a problem making this change, I was just attempting to mimic the actual GitLab error message. Should I still use a more descriptive message?

Reply:
Yes, the mock is primarily used for tests and I would be happier with a message that points the problem immediately, rather than trying to find the response in the doc.

Ideally, the API response would be more precise too ^^, `per_page has a value not allowed` is a bit vague